gnosis_vpn-lib/src/config/mod.rs

```diff
@@ -0,0 +1,101 @@
+use std::collections::HashMap;
+use std::fs;
+use std::path::Path;
+
+use thiserror::Error;
+
+use crate::connection::Destination;
+use crate::entry_node::EntryNode;
+use crate::peer_id::PeerId;
+use crate::wireguard::config::Config as WireGuardConfig;
+
+mod v1;
+mod v2;
+
+pub const DEFAULT_PATH: &str = "/etc/gnosisvpn/config.toml";
+pub const ENV_VAR: &str = "GNOSISVPN_CONFIG_PATH";
+
+#[derive(Clone, Debug)]
+pub enum Config {
+    V1(v1::Config),
+    V2(v2::Config),
+}
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("Configuration file not found")]
+    NoFile,
+    #[error("IO error: {0}")]
+    IO(#[from] std::io::Error),
+    #[error("Deserialization error: {0}")]
+    Deserialization(#[from] toml::de::Error),
+    #[error("Unsupported config version: {0}")]
+    VersionMismatch(u8),
+}
+
+pub fn read(path: &Path) -> Result<Config, Error> {
+    let content = fs::read_to_string(path).map_err(|e| {
+        if e.kind() == std::io::ErrorKind::NotFound {
+            Error::NoFile
+        } else {
+            Error::IO(e)
+        }
+    })?;
+
+    let table = content.parse::<toml::Table>()?;
+    let res_v2 = toml::from_str::<v2::Config>(&content);
+    match res_v2 {
+        Ok(config) => {
+            if config.version == 2 {
+                let wrong_keys = v2::wrong_keys(&table);
+                for key in wrong_keys.iter() {
+                    tracing::warn!("ignoring unsupported key in configuration file: {}", key);
```

Comment:
When logging, pls use structured logging with dynamic content being in fields and static content in the string:
```suggestion
                    tracing::warn!("ignoring unsupported key in configuration file: {}", key);
```
```suggestion
                    tracing::warn!(%key, "ignoring unsupported key in configuration file");
```